vms/platformvm/state/disk_staker_diff_iterator_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/thepudds/fzgen/fuzzer"
+
+	"github.com/ava-labs/avalanchego/database/memdb"
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+func FuzzMarshalDiffKey(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		require := require.New(t)
+
+		var (
+			subnetID ids.ID
+			height   uint64
+			nodeID   ids.NodeID
+		)
+		fz := fuzzer.NewFuzzer(data)
+		fz.Fill(&subnetID, &height, &nodeID)
+
+		key := marshalDiffKey(subnetID, height, nodeID)
+		parsedSubnetID, parsedHeight, parsedNodeID, err := unmarshalDiffKey(key)
+		require.NoError(err)
+		require.Equal(subnetID, parsedSubnetID)
+		require.Equal(height, parsedHeight)
+		require.Equal(nodeID, parsedNodeID)
+	})
+}
+
+func FuzzUnmarshalDiffKey(f *testing.F) {
+	f.Fuzz(func(t *testing.T, key []byte) {
+		require := require.New(t)
+
+		subnetID, height, nodeID, err := unmarshalDiffKey(key)
+		if err != nil {
+			require.ErrorIs(err, errUnexpectedDiffKeyLength)
+			return
+		}
+
+		formattedKey := marshalDiffKey(subnetID, height, nodeID)
+		require.Equal(key, formattedKey)
+	})
+}
+
+func TestDiffIteration(t *testing.T) {
```

Comment:
Should this test a set of potential edge cases or add property testing instead? This seems like it just covers one case.

Reply:
validator set diff iteration is already property tested

Reply:
ref: https://github.com/ava-labs/avalanchego/blob/master/vms/platformvm/validator_set_property_test.go